src/Filters/remez_fir.jl

```diff
@@ -0,0 +1,829 @@
+
+Copyright 2005 Travis Oliphant
+Permission to use, copy, modify, and distribute this software without fee
+is granted under the SciPy License.
+
+C CODE BANNER
+
+/********************************************************
+ *
+ *  Code taken from remez.c by Erik Kvaleberg which was 
+ *    converted from an original FORTRAN by:
+ *
+ * AUTHORS: JAMES H. MCCLELLAN
+ *
+ *         DEPARTMENT OF ELECTRICAL ENGINEERING AND COMPUTER SCIENCE
+ *         MASSACHUSETTS INSTITUTE OF TECHNOLOGY
+ *         CAMBRIDGE, MASS. 02139
+ *
+ *         THOMAS W. PARKS
+ *         DEPARTMENT OF ELECTRICAL ENGINEERING
+ *         RICE UNIVERSITY
+ *         HOUSTON, TEXAS 77001
+ *
+ *         LAWRENCE R. RABINER
+ *         BELL LABORATORIES
+ *         MURRAY HILL, NEW JERSEY 07974
+ *
+ *  
+ *  Adaptation to C by 
+ *      egil kvaleberg
+ *      husebybakken 14a
+ *      0379 oslo, norway
+ *  Email:
+ *      egil@kvaleberg.no
+ *  Web:
+ *      http://www.kvaleberg.com/
+ *
+ *********************************************************/
+
+=============================================#
+
+#using Formatting
+
+# RemezFilterType:
+#    Type I and II symmetric linear phase: neg==0   (filter_type==bandpass)
+#    Type III and IV negative symmetric linear phase: neg==1   (filter_type==hilbert or differentiator)
+@enum RemezFilterType filter_type_bandpass=1 filter_type_differentiator=2 filter_type_hilbert=3
+
+
+
+"""/*
+ *-----------------------------------------------------------------------
+ * FUNCTION: lagrange_interp (d)
+ *  FUNCTION TO CALCULATE THE LAGRANGE INTERPOLATION
+ *  COEFFICIENTS FOR USE IN THE FUNCTION gee.
+ *-----------------------------------------------------------------------
+ */"""
+function lagrange_interp(k::Integer, n::Integer, m::Integer, x::AbstractVector)
+    retval = 1.0
+    q = x[k]
+    for l = 1 : m
+        for j = l : m : n
+            if j != k
+                retval *= 2.0 * (q - x[j])
+            end
+        end
+    end
+    1.0 / retval
+end
+
+
+"""
+eff(freq::Float64, fx::AbstractVector, lband::Integer, filter_type::RemezFilterType)
+
+/*
+ *-----------------------------------------------------------------------
+ * FUNCTION: eff
+ *  FUNCTION TO CALCULATE THE DESIRED MAGNITUDE RESPONSE
+ *  AS A FUNCTION OF FREQUENCY.
+ *  AN ARBITRARY FUNCTION OF FREQUENCY CAN BE
+ *  APPROXIMATED IF THE USER REPLACES THIS FUNCTION
+ *  WITH THE APPROPRIATE CODE TO EVALUATE THE IDEAL
+ *  MAGNITUDE.  NOTE THAT THE PARAMETER FREQ IS THE
+ *  VALUE OF NORMALIZED FREQUENCY NEEDED FOR EVALUATION.
+ *-----------------------------------------------------------------------
+ */
+"""
+function eff(freq::Float64, fx::AbstractVector, lband::Integer, filter_type::RemezFilterType)
+    if filter_type == filter_type_differentiator
+        return fx[lband] * freq
+    else 
+        return fx[lband]
+    end
+end
+
+"""
+wate(freq::Float64, fx::AbstractVector, wtx::AbstractVector, lband::Integer, filter_type::RemezFilterType)
```

Comment:
Not that it matters much, but technically, the signature here is missing the `wtx` parameter.